### ISSUE TITLE
marker to show search results that have been viewed only works if Return to Records button is used (ericb)

### ISSUE THREAD

**Issue by FreeUKGenIssues** [FreeUKGen/FreeUKRegProductIssues#135]
Issue reported by **ericb** at 2015-01-02 17:09:28 UTC
Time: 2015-01-02T17:06:05+00:00
Session ID: f6852b1bd8afe3477ee0b49e052421b4
Problem Page URL: [/freereg1_csv_files/5463b40be9379010fc098149](/freereg1_csv_files/5463b40be9379010fc098149)
Previous Page URL: [http://test2.freereg.org.uk/places/53cd4998eca9ebee4f00b86a](http://test2.freereg.org.uk/places/53cd4998eca9ebee4f00b86a)
Reported Issue:
I have discovered that the feature to add marker to show "Viewed" against each item when one looks up the details, only works when one uses the "tab" to return to results. and NOT when one uses the Browser back button.

Which would be OK if we were consistent in always having a button to navigate back with. But that's not the case. Most of the way through the Members area, one has to navigate backwards by Browser back button - the tab to do this which was there on some screens, seems to have gone now?

So request is to be consistent please.    

![Screenshot](http://test2.freereg.org.uk)

**Comment by Captainkirkdawson** [comment]
The viewed tag is set every time you view the detailed result for an active or stored search. 
What you are seeing is a result of local page caching. When you use the back button your browser redisplayed the page as stored in your local cache of that page. i.e. the index of results in this case.
All browsers do this to make your experience seem faster. 
If you were to tell the browser to reload the page then you would see the viewed tag.
The Back to Results option on the details page tells the browser to redisplay the results index page which will show the viewed tag..
You can tell the browser to not cache pages by changing your internet settings, This will impact the responsiveness of your system. 
Alternatively we tell your Browser not to cache the page which will impact the responsiveness for all users.
Bottom line is that the system is working as it should.

WRT to your other comments about back in general. In almost all cases the Back button on earlier displays did nothing more than the Browser back button so it seemed a waste of space to add an additional button for a capability that already existed on the page.

In many cases simply going back to the cached page is exactly what we as users want to do, Since if it is a form we are submitting going back to the page gives you the entries you had selected. If we do a redisplay form we have to do a fair amount of work to redisplay your previous entries and a new form option as well.If there are specific aspects of this as they relate to member pages can we please put that into a separate story.

**Comment by Captainkirkdawson** [comment]
The viewed flag is set for every visit to a detail record while the search for which it is a member is active regardless of how you leave the display.
If you use the browser back button then most browser will simply redisplay the page from local cache on your computer. It does not normally ask the server to redisplay the page, If your cache has expired for some reason the browser may get a copy from an intermediate source on the internet.
This means that the viewed tag will not be shown as it is not in the cache. You can refresh the display by the reload page on your browser. The viewed tag will then become apparent. Sometimes you may even be required to do a cntl reload to force the browser to reload.
If you use the Return to Results  button the server actually tells the Browser to refresh the page so you will see the viewed immediately.
In summary the system is doing what it should, The only way to achieve the browser back doing a refresh straight away is for you to stop your local caching which as implications for your bandwidth usage and responsiveness. Seer the caching options for your internet options;Or we tell your browser not to cache and that will have also issues off slowing down the display of your pages

**Comment by Sherlock21** [comment]
Thank you for that detailed reply.
I understand your  explanation - indeed, if I had thought carefully, i already knew the differences.
 I think my problem arose from my own lack error - that of falling into auto pilot mode, and just got into the browser back button nav.  Your green button to Return is a big button, i agree, but the text in it is too feint to read at a glance. Especially as in some screens where there are more than one of them, they all merge into one long line.

SO I agree, what you have provided is just fine, if you could sharpen that font without enlarging the button or taking more vertical space that would help.

Nothing else need be done for this one then.

**Comment by Captainkirkdawson** [comment]
Comment added to public search page review
